packages/cli/src/hooks/init.ts

```diff
@@ -1,23 +1,44 @@
 import {Hook} from '@oclif/core'
+import {sniffForPath} from '@shopify/cli-kit/node/custom-oclif-loader'
+import {fileExistsSync} from '@shopify/cli-kit/node/fs'
+import {cwd, joinPath} from '@shopify/cli-kit/node/path'
 import {renderWarning} from '@shopify/cli-kit/node/ui'
+import {readFileSync} from 'fs'
 
+/**
+ * When running a Hydrogen command (except hydrogen:init), check if the current directory is a Hydrogen project.
+ * If it's not, show a warning and exit. We need to do this here because some hydrogen commands depend on some peerDependencies
+ * that are loaded dynamically from the project's package.json.
+ *
+ * To check if the project is a Hydrogen one, we look for the presence of the \@shopify/hydrogen dependency in the project's package.json.
+ *
+ * @param options - The options passed to the hook.
+ */
 const hook: Hook<'init'> = async (options) => {
```

Comment:
Should this be part of the hydrogen OCLIF package? Rather than CLI?

Reply:
Good point, we needed this here before because we needed to check the case where you didn't have the hydrogen plugin, but it can be moved to hydrogen repo now. I'll do that for the next release though as it's not breaking anything